File: Coins2/app/src/main/java/com/lq/coins/RecyclerAdapter.java
package com.lq.coins;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lq.coins.databinding.RecyclerItemBinding;

import java.util.ArrayList;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.AdapterViewHolder> {
     String [] colors = {"#edd000","#054170","#664499","#66cdaa","#ea9782","#559933","#e1d4c0","#e97042"} ;
     ArrayList<CoinModel> items ;
     RecyclerAdapter(ArrayList<CoinModel> items)
     {
         this.items = items;
     }

    @NonNull
    @Override
    public AdapterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
       RecyclerItemBinding binding = RecyclerItemBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new RecyclerAdapter.AdapterViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull AdapterViewHolder holder, int position) {
              holder.binding.coinNameTextView.setText(items.get(position).coinName);
              holder.binding.coinPriceNameTextView.setText(items.get(position).coinPair);
              holder.itemView.setBackgroundColor(Color.parseColor(colors[position % 7 ]));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static  class AdapterViewHolder extends RecyclerView.ViewHolder{
          RecyclerItemBinding binding;
         public AdapterViewHolder(RecyclerItemBinding binding) {
             super(binding.getRoot());
             this.binding = binding;
         }
     }
}

File: CryptoAppWithKotlin/app/src/main/java/com/lq/cryptoappwithkotlin/utulitys/RecyclerAdapter.kt
package com.lq.cryptoappwithkotlin.utulitys

import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.lq.cryptoappwithkotlin.R
import com.lq.cryptoappwithkotlin.databinding.RecyclerItemBinding
import com.lq.cryptoappwithkotlin.modal.CryptoModal

class RecyclerAdapter(val items : ArrayList<CryptoModal>,val colorArr:List<String>) : RecyclerView.Adapter<RecyclerAdapter.AdapterHolder>() {

      class AdapterHolder (val binding : RecyclerItemBinding) : ViewHolder(binding.root)
      {

      }

      override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AdapterHolder {
          val binding : RecyclerItemBinding = RecyclerItemBinding.inflate(LayoutInflater.from(parent.context),parent,false);
          return AdapterHolder(binding);
      }

      override fun getItemCount(): Int {
           return items.size
      }

      override fun onBindViewHolder(holder: AdapterHolder, position: Int) {
            holder.itemView.setBackgroundColor(Color.parseColor(colorArr.get(position % 8)))
            holder.binding.cryptoNameTextView.text = "CRYPTO : ${items[position].currency}";
            holder.binding.cryptoPriceTextView.text ="PRICE : ${items[position].price}";
            holder.binding.imageView.setImageResource(R.drawable.crypto)
            holder.itemView.setOnClickListener {
                  AlertDialog.Builder(holder.itemView.context)
                        .setTitle("CRYPTO : ${items[position].currency}")
                        .setMessage("Price : ${items[position].price}")
                        .setNegativeButton("CLOSE") { dialogInterface, i ->
                               dialogInterface.cancel();
                         }
                        .show();
            }
      }
}